tests/dofs/dof_tools_21_b.cc

```diff
@@ -317,37 +317,29 @@ print_matching(DoFHandler<dim, spacedim> &dof_handler,
   deallog << std::endl;
 
 
-  std::bitset<3> orientation;
-  if (not GridTools::orthogonal_equality(orientation,
-                                         face_1,
-                                         face_2,
-                                         dim == 2 ? 1 : 2,
-                                         dealii::Tensor<1, spacedim>()))
-    std::cerr << " not match! oh noze!! " << std::endl;
-  deallog << "Orientation: " << orientation[0] << orientation[1]
-          << orientation[2] << std::endl;
-
-
-  DoFTools::make_periodicity_constraints(face_1,
-                                         face_2,
-                                         constraint_matrix,
-                                         velocity_mask,
-                                         orientation[0],
-                                         orientation[1],
-                                         orientation[2]);
+  const auto orientation = GridTools::orthogonal_equality(
+    face_1, face_2, dim == 2 ? 1 : 2, Tensor<1, spacedim>());
+  AssertThrow(orientation, ExcMessage(" not match! oh noze!! "));
```

Comment:
:-)

But that's not a useful thing for the user. I think you should use `ExcInternalError()` here and below.

Reply:
Oh fine. It was pre-existing :)